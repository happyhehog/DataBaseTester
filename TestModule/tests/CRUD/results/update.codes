{"status":"OK","command":"SELECT * from t;","answer":"id|height  |nickname  |num|\n5 |5.000000|'5'       |200|\n6 |6.000000|'6'       |200|\n7 |7.000000|'7'       |200|\n8 |8.000000|'8'       |200|\n9 |9.000000|'9'       |200|\n1 |2.000000|'nickname'|100|\n2 |2.000000|'nickname'|100|\n3 |2.000000|'nickname'|100|\n4 |2.000000|'nickname'|100|\n10|0.000000|'nickname'|200|\n11|0.000000|'nickname'|200|\n12|0.000000|'nickname'|200|\n13|0.000000|'nickname'|200|\n14|0.000000|'nickname'|200|\n15|0.000000|'nickname'|200|\n16|0.000000|'nickname'|200|\n17|0.000000|'nickname'|200|\n18|0.000000|'nickname'|100|\n19|0.000000|'nickname'|100|\n41|0.000000|'nickname'|100|"}
