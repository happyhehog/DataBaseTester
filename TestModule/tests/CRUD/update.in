[@Clear]
[@PrintLevel] MAIN

@@ ==============
@@ Testing UPDATE
@@ ==============

[@NoOutput]
CREATE TABLE t(id INT NOT NULL PRIMARY KEY, height FLOAT, nickname CHAR(20), num INT);

[#Repeat] 20
INSERT INTO t VALUES ($i, $i, $i, $i);

UPDATE t SET num = 200;
UPDATE t SET num = 100, height = 2.0 WHERE id <= 4 or id >= 18;
UPDATE t SET height = 0 WHERE id >= 9.9 or id = 20;
UPDATE t SET nickname = 'nickname' WHERE height = 0 or not (num = 200);
UPDATE t SET id = 20 * 2 + 1 WHERE id = 20;

[@NoOutput]

SELECT * from t;

[@NoOutput]
DROP TABLE t;
[@NoOutput]