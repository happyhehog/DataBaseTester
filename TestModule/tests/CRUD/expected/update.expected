id|height  |nickname  |num|
5 |5.000000|'5'       |200|
6 |6.000000|'6'       |200|
7 |7.000000|'7'       |200|
8 |8.000000|'8'       |200|
9 |9.000000|'9'       |200|
1 |2.000000|'nickname'|100|
2 |2.000000|'nickname'|100|
3 |2.000000|'nickname'|100|
4 |2.000000|'nickname'|100|
10|0.000000|'nickname'|200|
11|0.000000|'nickname'|200|
12|0.000000|'nickname'|200|
13|0.000000|'nickname'|200|
14|0.000000|'nickname'|200|
15|0.000000|'nickname'|200|
16|0.000000|'nickname'|200|
17|0.000000|'nickname'|200|
18|0.000000|'nickname'|100|
19|0.000000|'nickname'|100|
41|0.000000|'nickname'|100|
